io/input/line_inputformat.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "io/input/line_inputformat.hpp"
+
+#include <string>
+
+#include "boost/utility/string_ref.hpp"
+
+#include "core/utils.hpp"
+
+#include "io/input/inputformat_helper.hpp"
+
+namespace husky {
+namespace io {
+
+enum LineInputFormatSetUp {
+    NotSetUp = 0,
+    InputSetUp = 1 << 1,
+    AllSetUp = InputSetUp,
+};
+
+LineInputFormat::LineInputFormat() { is_setup_ = LineInputFormatSetUp::NotSetUp; }
+
+LineInputFormat::~LineInputFormat() {
+    ASSERT_MSG(splitter_ != nullptr, "Trying to destroy an illegal LineInputFormat object");
+    delete splitter_;
+    splitter_ = nullptr;
+}
+
+bool LineInputFormat::is_setup() const { return !(is_setup_ ^ LineInputFormatSetUp::AllSetUp); }
+
+void LineInputFormat::set_input(const std::string & url) {
+    ASSERT_MSG(!is_setup(), "Last input not finished");
+    int prefix = url.find("://");

```

Comment:
Since there're many same part like this to parse and construct the Splitter, I think these lines of code can be integrated into a function, which can be called by these inputformats.